src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceParameterService.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.IntroduceVariable;
+using Microsoft.CodeAnalysis.Simplification;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.IntroduceVariable
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp), Shared]
+    internal partial class CSharpIntroduceParameterService : AbstractIntroduceParameterService<
+        CSharpIntroduceParameterService,
+        ExpressionSyntax,
+        InvocationExpressionSyntax,
+        IdentifierNameSyntax>
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpIntroduceParameterService()
+        {
+        }
+
+        protected override async Task<Solution> IntroduceParameterAsync(SemanticDocument document, ExpressionSyntax expression, bool allOccurrences, bool trampoline, CancellationToken cancellationToken)
+        {
+            if (!trampoline)
+            {
+                return await IntroduceParameterForRefactoringAsync(document, expression, allOccurrences, cancellationToken).ConfigureAwait(false);
+
+            }
+            else
+            {
+                return await IntroduceParameterForTrampolineAsync(document, expression, allOccurrences, cancellationToken).ConfigureAwait(false);
+            }
+        }
+
+        protected override SeparatedSyntaxList<SyntaxNode> AddArgumentToArgumentList(SeparatedSyntaxList<SyntaxNode> invocationArguments, SyntaxNode newArgumentExpression)
+        {
+            return invocationArguments.Add(SyntaxFactory.Argument((ExpressionSyntax)newArgumentExpression.WithoutAnnotations(ExpressionAnnotationKind).WithAdditionalAnnotations(Simplifier.Annotation)));
```

Comment:
what's teh WithoutAnnotations bit for?

Reply:
We do not want the expression that is being introduced as an argument at the callsite to have the annotation being used to locate the original expression in the tree.

Reply:
1. doc please.
2. i don't really understand.  you're going to have to *synthesize* an expression at the callsite.  so how would it have an annotatino from the original expr?
3. you should have the *real* unannotated original expr from the start.  can you not just use that?  instead of annotating it, then de-annotating it?

Reply:
On your third point, because I am using an updated semantic document with the expression replaced with a annotated expression, if I just pass in the original expression it can't find it in the syntax tree
If I do not remove the annotation in the synthesized expression, then the annotation persists and that argument has an incorrect annotation on it.